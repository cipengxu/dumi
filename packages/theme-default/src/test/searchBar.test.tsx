import '@testing-library/jest-dom';
import React from 'react';

import { highlight } from '../components/SearchBar';

describe('test highlight', () => {
  it('should render right', () => {
    expect(highlight('', '1234')).toMatchSnapshot();
    expect(highlight('1', '1234')).toMatchSnapshot();
    expect(highlight('2', '1234')).toMatchSnapshot();
    expect(highlight('23', '1234')).toMatchSnapshot();
  });
});
