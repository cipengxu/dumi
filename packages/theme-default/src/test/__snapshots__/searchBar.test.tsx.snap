// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<React.Fragment>
  <span
    className="__dumi-default-search-highlight"
  >
    
  </span>
  <span>
    1234
  </span>
</React.Fragment>
`;

exports[` 2`] = `
<React.Fragment>
  <span
    className="__dumi-default-search-highlight"
  >
    1
  </span>
  <span>
    234
  </span>
</React.Fragment>
`;

exports[` 3`] = `
<React.Fragment>
  <span>
    1
  </span>
  <span
    className="__dumi-default-search-highlight"
  >
    2
  </span>
  <span>
    34
  </span>
</React.Fragment>
`;

exports[` 4`] = `
<React.Fragment>
  <span>
    1
  </span>
  <span
    className="__dumi-default-search-highlight"
  >
    23
  </span>
  <span>
    4
  </span>
</React.Fragment>
`;

exports[`test highlight should render right 1`] = `
<React.Fragment>
  <span
    className="__dumi-default-search-highlight"
  >
    
  </span>
  <span>
    1234
  </span>
</React.Fragment>
`;

exports[`test highlight should render right 2`] = `
<React.Fragment>
  <span
    className="__dumi-default-search-highlight"
  >
    1
  </span>
  <span>
    234
  </span>
</React.Fragment>
`;

exports[`test highlight should render right 3`] = `
<React.Fragment>
  <span>
    1
  </span>
  <span
    className="__dumi-default-search-highlight"
  >
    2
  </span>
  <span>
    34
  </span>
</React.Fragment>
`;

exports[`test highlight should render right 4`] = `
<React.Fragment>
  <span>
    1
  </span>
  <span
    className="__dumi-default-search-highlight"
  >
    23
  </span>
  <span>
    4
  </span>
</React.Fragment>
`;
